R/learner_qgam_regr_qgam.R

```diff
@@ -0,0 +1,142 @@
+#' @title Regression Quantile Generalized Additive Model Learner
+#' @author lona-k
+#' @name mlr_learners_regr.qgam
+#'
+#' @description
+#' Quantile Regression with generalized additive models.
+#' Calls [qgam::qgam()] from package \CRANpkg{qgam}.
+#'
+#' @section Form:
+#' For the `form` parameter, a gam formula specific to the [Task][mlr3::Task] is required
+#' (see example and `?mgcv::formula.gam`).
+#' If no formula is provided, a fallback formula using all features in the task is used that will make the Learner
+#' behave like Linear Quantile Regression.
+#' Only features specified in the formula are used, superseding columns with col_roles "feature" in the task.
```

Comment:
Maybe we can check that the formula does not contain more features than specified in the column role?

Reply:
Are that they are the same?